app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -0,0 +1,74 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.RatioExpression
+
+/** This class contains method that helps to parse string to ratio. */
+class StringToRatioParser {
+  private val invalidCharsRegex =
+    """^[\d\s/\:\.]+$""".toRegex()
+  private val invalidRatioRegex =
+    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
```

Comment:
Here & elsewhere: I suggest normalizing the string first to simplify the whitespace checks (see how fractions does it). I think you can take this one step farther and strip all whitespace after normalizing to simplify both the format verification & parsing steps.

Reply:
Actually as discussed in meeting, removing whitespace probably won't work during verification but normalization should still simplify things.

Reply:
i think we discussed this thing that if the number entered is something like 1:2   4:5 it that would be considered as a valid ratio and that isn't correct. Does that make sense?

Reply:
Yes we did discuss it, and as I mentioned above it doesn't make sense to remove whitespace for validation, but it still makes sense to normalize. Right now, your regex pattern is matching against a variable number of white space characters--that isn't needed if you normalize. See the fractions parser for an example of how the pattern can be simplified by using normalize whitespace.

Also, I'm suggesting that you normalize, _not_ remove. '1:2 4:5' will still be validated correctly when you use normalize whitespace.